src/Elastic.Transport/DefaultHttpTransport.cs

```diff
@@ -185,28 +185,28 @@ public DefaultHttpTransport(TConfiguration configurationValues) : this(configura
 			Settings.OnRequestDataCreated?.Invoke(requestData);
 			TResponse response = null;
 
-			if (OpenTelemetry.ElasticTransportActivitySource.HasListeners() && activity.IsAllDataRequested)
+			if (activity is { IsAllDataRequested: true })
```

Comment:
```suggestion
			if (OpenTelemetry.ElasticTransportActivitySource.HasListeners() && activity is { IsAllDataRequested: true })
```

Reply:
This is not necessary. If `OpenTelemetry.ElasticTransportActivitySource.HasListeners()` is false, `activity` must be null.

Reply:
Ahh missed the earlier assignment to `activity`